### PR TITLE
VPA: Store annotation metadata in internal structure

### PR DESCRIPTION
**Pull request by maltenbergert**

### CODE REVIEW

**Comment by maltenbergert** [comment]
Ready for review.

**Comment by maltenbergert** [comment]
Ready for review.

File: vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go [hunk 1]
```diff
@@ -299,6 +301,22 @@ func TestChangePodLabels(t *testing.T) {
...
+	// Verify that the annotations match the test annotations.
+	assert.Equal(t, vpa.Annotations, testAnnotations)
+	// Update the annotations (non-empty).
+	annotations := vpaAnnotationsMap{"key-2": "value-2"}
+	vpa = addVpa(cluster, testVpaID, annotations, testSelectorStr)
+	assert.Equal(t, vpa.Annotations, annotations)
+	// Update the annotations (empty).
+	annotations = vpaAnnotationsMap{}
+	vpa = addVpa(cluster, testVpaID, annotations, testSelectorStr)
+	assert.Equal(t, vpa.Annotations, annotations)
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go]: nit: I'd rather directly compare with empty vpaAnnotationsMap

**Pull request merged**